### ISSUE TITLE
Speaker embedding never selected during synthesis

### ISSUE THREAD

**Issue by McTi** [coqui-ai/TTS#932]
**Describe the bug**
When running synthesis for a multi-speaker model that uses speaker_embedding, d_vectors are selected instead causing a `TypeError: 'int' object is not subscriptable `
Found using Tacotron2, but it seems like it would affect other models.

```
Traceback (most recent call last):
  File "/home/anaconda3/envs/coqui/bin/tts", line 33, in <module>
    sys.exit(load_entry_point('TTS', 'console_scripts', 'tts')())
  File "/home/Coqui/TTS/TTS/bin/synthesize.py", line 226, in main
    synthesizer = Synthesizer(
  File "/home/Coqui/TTS/TTS/utils/synthesizer.py", line 72, in __init__
    self._load_tts(tts_checkpoint, tts_config_path, use_cuda)
  File "/home/Coqui/TTS/TTS/utils/synthesizer.py", line 110, in _load_tts
    speaker_manager = self._init_speaker_manager()
  File "/home/Coqui/TTS/TTS/utils/synthesizer.py", line 132, in _init_speaker_manager
    speaker_manager = SpeakerManager(d_vectors_file_path=self.tts_speakers_file)
  File "/home/Coqui/TTS/TTS/tts/utils/speakers.py", line 77, in __init__
    self.set_d_vectors_from_file(d_vectors_file_path)
  File "/home/Coqui/TTS/TTS/tts/utils/speakers.py", line 164, in set_d_vectors_from_file
    speakers = sorted({x["name"] for x in self.d_vectors.values()})
  File "/home/Coqui/TTS/TTS/tts/utils/speakers.py", line 164, in <setcomp>
    speakers = sorted({x["name"] for x in self.d_vectors.values()})
TypeError: 'int' object is not subscriptable
```
**To Reproduce**
$ tts --config_path config.json --model_path checkpoint_60000.pth.tar --text "Some random text." --speaker_idx speaker1 --speakers_file_path speakers.json

**Expected behavior**
With use_speaker_embedding "true" and "use_d_vector_file "false",  speaker embedding should be selected.

**Environment (please complete the following information):**
- Linux Ubuntu 20.04.3
- PyTorch 1.10.0
- Python 3.9.7

**Additional context**
It looks like the following code will always select a `d_vector_file ` even if  `self.tts_config.use_speaker_embedding is True`

https://github.com/coqui-ai/TTS/blob/33aa27e2d6a80d38a7ea8928ad14d032d2c53a40/TTS/utils/synthesizer.py#L117-L132

[config.txt](https://github.com/coqui-ai/TTS/files/7530611/config.txt)

**Comment by erogol** [comment]
Solved it ? @McTi

**Comment by McTi** [comment]
I closed it because it's a duplicate of #919 

As a workaround for me, I changed the second chunk of code to be:

```
     if hasattr(self.tts_config, "use_d_vector_file") and self.tts_config.use_d_vector_file is True: 
         if self.tts_speakers_file: 
             speaker_manager = SpeakerManager(d_vectors_file_path=self.tts_speakers_file) 
         if self.tts_config.get("d_vector_file", None): 
             speaker_manager = SpeakerManager(d_vectors_file_path=self.tts_config.d_vector_file) 
     return speaker_manager 

```

But I'm not familiar enough with the intended use of d_vector_files to tell whether that's a valid general fix.

**Comment by skol101** [comment]
@McTi is correct that this code has to be changed in order for inference to work on multi speaker model. 

Strange thing is here: even if _use_d_vector_file_ param is removed from the config.json ( btw, this param is presented in two places in the config.json), the code in synthesizer.py will still execute this part

```
if hasattr(self.tts_config, "use_d_vector_file") and self.tts_config.use_d_vector_file is True: 
         if self.tts_speakers_file: 
             speaker_manager = SpeakerManager(d_vectors_file_path=self.tts_speakers_file) 
```
